granulate_utils/metadata/cloud.py

```diff
@@ -167,9 +168,9 @@ def get_static_cloud_instance_metadata(logger: Union[logging.LoggerAdapter, logg
     def _fetch() -> Tuple[Optional[Metadata], List[Exception]]:
         cloud_metadata_fetchers = [get_aws_metadata, get_gcp_metadata, get_azure_metadata]
         raised_exceptions: List[Exception] = []
-        for fetcher in cloud_metadata_fetchers:
+        for future in call_in_parallel(cloud_metadata_fetchers, timeout=6):
             try:
-                response = fetcher()
+                response = future.result()
```

Comment:
Aren't the other futures leaked now if the first completed one was successful?

Reply:
They are leaked until they finish running in the background, which is why I documented that it should be ok as long as the callables are limited in time.

In terms of object/memory leak, I don't think that's the case:
```
In [13]: len(gc.get_objects())
Out[13]: 142881

In [14]: next(call_in_parallel([lambda: 1 for _ in range(1000000)], timeout=10)).result()
Out[14]: 1

In [15]: len(gc.get_objects())
Out[15]: 141374

In [16]: next(call_in_parallel([lambda: i for i in range(1000000)], timeout=10)).result()
Out[16]: 999999

In [17]: len(gc.get_objects())
Out[17]: 141410
```

Reply:
It's okay after the TPE and threads die